include/circt/Support/FVInt.h

```diff
@@ -646,8 +646,8 @@ inline FVInt operator-(uint64_t a, const FVInt &b) {
 
 inline FVInt operator-(const APInt &a, const FVInt &b) { return FVInt(a) - b; }
 
-inline bool operator==(uint64_t a, const FVInt &b) { return b == a; }
-inline bool operator!=(uint64_t a, const FVInt &b) { return b != a; }
+inline bool operator==(uint64_t a, const FVInt &b) { return b.operator==(a); }
```

Comment:
How is this related to the slang version? Is it picking up some overzealous global operator in slang somewhere?

Reply:
This is a consequence of building `circt` code with C++ 20 for the first time, where https://timsong-cpp.github.io/cppwp/n4868/over.match#oper-3.4.4 has come into play. If my analysis is correct, operator resolution of `==` in `{ return b == a; }` where `a: uint64_t` and `b: const FVInt &b` resolved to `inline bool operator==(uint64_t a, const FVInt &b) { return b == a; }` , i.e. itself, leading to infinite recursion.

Reply:
I see, just an artifact of C++20 and not anything slang is doing. Maybe you can just ifdef out these overloads when compiling in C++20 mode.